Svc/TlmChan/TlmChanImplRecv.cpp

```diff
@@ -38,7 +38,9 @@ namespace Svc {
                     FW_ASSERT(this->m_tlmEntries[this->m_activeBuffer].free < TLMCHAN_HASH_BUCKETS);
                     // add new bucket from free list
                     entryToUse = &this->m_tlmEntries[this->m_activeBuffer].buckets[this->m_tlmEntries[this->m_activeBuffer].free++];
-                    prevEntry->next = entryToUse;
+                    if(prevEntry) {
```

Comment:
If the code assumes this exists, shouldn't this be an assert not an if guard?

Reply:
Agreed, an assertion is a better approach. The static analyzer is false warning on a prevEntry being null case which isn't smart enough to realize that entryToUse is guaranteed non-null the first time through the loop.

Reply:
So you'll change this to an assert, or this is still there to appease the static analyzer?